frontend/public/components/factory/list-page.jsx

```diff
@@ -183,6 +183,16 @@ export const FireMan_ = connect(null, {filterList})(
       params.forEach((v, k) => this.applyFilter(k, v));
     }
 
+    runOrNavigate = (itemName) => {
+      const createProps = this.props.createProps || {};
+      if (createProps.action && createProps.action(itemName)) {
```

Comment:
Why are we calling `createProps.action` twice? Duplicity can (and usually does) lead to bugs.

Why can't we strive for the (arguably) best code possible whenever adding new code? :smiley: 

```ts
if (_.isFunction(createProps.action)) {
  createProps.action(itemName)();
} else if (_.isFunction(createLink)) {
  history.push(createProps.createLink(itemName));
}
```

Reply:
completely agree, this duplicity was added during addressing other review comment. Will change it back.

Reply:
done